### PR TITLE
Refactor names

### PR DESCRIPTION
**Pull request by spex66**
- fixed a forgotten interface still having the `dry_run` parameter
-  removed 'transformation' acl from `acl_all_scope_only_types`
    as it now supports dataset scopes too!
- refactor variable names to match the new documentation
   1. `group_types_dimensions` > `group_bootstrap_hierarchy`
   2. `group_type` > `group_ns` (**namespace**: src, ca, uc)
   3. `group_prefix` > `group_core` (src:001:sap)
- working on a test-config `tests/example/config-deploy-bootstrap.yml`

### CODE REVIEW

**Comment by sdorheim** [comment]
Instead of group_core I suggest group_unit or group_entity

File: incubator/bootstrap_cli/__main__.py [hunk 1]
```diff
@@ -202,14 +213,16 @@ class BootstrapCore:
...
+    AGGREGATED_GROUP_NAME = "allprojects"
 
     def __init__(self, configpath: str, delete: bool = False):
         if delete:
             self.config: BootstrapDeleteConfig = BootstrapDeleteConfig.from_yaml(configpath)
             self.delete_or_deprecate: Dict[str, Any] = self.config.delete_or_deprecate
         else:
             self.config: BootstrapDeployConfig = BootstrapDeployConfig.from_yaml(configpath)
-            self.group_types_dimensions: Dict[str, Any] = self.config.bootstrap
+            self.group_bootstrap_dimensions: Dict[str, Any] = self.config.bootstrap
```
Comment (sdorheim) [inline incubator/bootstrap_cli/__main__.py:225]: This naming is still unclear to me. Can we come up with a better name?

File: incubator/bootstrap_cli/__main__.py [hunk 2]
```diff
@@ -240,20 +253,20 @@ def acl_template(actions, scope):
...
-        return f"{group_type}:allprojects" if group_type else "allprojects"
+    def get_allprojects_name_template(group_ns=None):
+        return f"{group_ns}:{BootstrapCore.AGGREGATED_GROUP_NAME}" if group_ns else BootstrapCore.AGGREGATED_GROUP_NAME
 
     @staticmethod
     def get_dataset_name_template():
         # 211013 pa: remove env prefixes
-        # return f"{shared_global_config['env']}:" + "{group_prefix}:dataset"
-        return "{group_prefix}:dataset"
+        # return f"{shared_global_config['env']}:" + "{group_core}:dataset"
```
Comment (sdorheim) [inline incubator/bootstrap_cli/__main__.py:262]: is this still needed, or can it be removed?

File: incubator/bootstrap_cli/__main__.py (hunk 1 above)
Comment (spex66) [reply incubator/bootstrap_cli/__main__.py:225]: would love to, but atm this is 1:1 the `bootstrap` config part from yaml
- as such it represents the namespace>core "dimensions"
- maybe `group_**_hierarchy` reflects better the topology of the "dimension"?

File: incubator/bootstrap_cli/__main__.py (hunk 2 above)
Comment (spex66) [reply incubator/bootstrap_cli/__main__.py:262]: the comment? I've tried to cleanup the old `shared_global_config` references, but missed that

File: incubator/bootstrap_cli/__main__.py (hunk 1 above)
Comment (sdorheim) [reply incubator/bootstrap_cli/__main__.py:225]: with `group_**_hierarchy` I would directly understand what is meant, but that might be just me

File: incubator/bootstrap_cli/__main__.py (hunk 2 above)
Comment (sdorheim) [reply incubator/bootstrap_cli/__main__.py:262]: Is it ok If I remove the rest and update the PR before merging?

File: incubator/bootstrap_cli/__main__.py (hunk 1 above)
Comment (spex66) [reply incubator/bootstrap_cli/__main__.py]: agreed and changed

**Review by sdorheim — APPROVED** [review]

**Pull request merged**